File: solutions/fortran/two-fer/1/two_fer.f90
module two_fer
  implicit none

contains

  function twoFer(name) result(phrase)
    character(*), intent(in), optional :: name
    character(:), allocatable :: phrase

    if (.not. present(name)) then
      phrase = "One for you, one for me."
    else
      phrase = "One for "//name//", one for me."
    end if
  end function twoFer

end module two_fer

File: solutions/c/two-fer/1/two_fer.c
#include "two_fer.h"
#include <string.h>
#include <stdlib.h>

void two_fer(char *buffer, const char *name){
    strcpy(buffer,"One for ");
    name == NULL ? strcat(buffer, "you") : strcat(buffer, name);
    strcat(buffer,", one for me.");
}
